modules/user_accounts/php/NDB_Form_user_accounts.class.inc

```diff
@@ -737,7 +737,12 @@ class NDB_Form_User_Accounts extends NDB_Form
 
         //------------------------------------------------------------
 
+        // checking if the account can be rejected (no password hash
+        // pending approval)
+        $this->tpl_data['can_reject'] =$this->_canRejectAccount($this->identifier);
```

Comment:
This only does the checks before adding the button on the frontend, it needs to also be checked on the backend before performing the deletion.